### PR TITLE
CI: continue on failure

### LINKED ISSUE #426: h5md_parallel test fails
```
77/85 Test #76: h5md_parallel ........................***Failed    2.07 sec
.F
======================================================================
FAIL: test_reload_dump (__main__.TestH5MD)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/espressopp/espressopp/testsuite/io/test_h5md_parallel.py", line 86, in test_reload_dump
    self.binary_compare('reference.h5', 'dump.h5')
  File "/__w/espressopp/espressopp/testsuite/io/test_h5md_parallel.py", line 35, in binary_compare
    self.assertEqual(b1, b2)
AssertionError: b'\x0[709 chars]00\x073Rb\x00\x00\x80\x00\x00\x00\x00\x00\x00\[1241 chars]\x00' != b'\x0[709 chars]00\x083Rb\x00\x00\x80\x00\x00\x00\x00\x00\x00\[1241 chars]\x00'

----------------------------------------------------------------------
Ran 2 tests in 0.051s

FAILED (failures=1)
Cubicity check passed -> powered by HeSpaDDA
HeSpaDDA message: Size Lenghts are eq. while ordering axis with preference on X, Y and Z!
nodeGrid:  (1, 1, 1)  cellGrid:  (7, 7, 7) half cell:  1
Cubicity check passed -> powered by HeSpaDDA
HeSpaDDA message: Size Lenghts are eq. while ordering axis with preference on X, Y and Z!
nodeGrid:  (1, 1, 1)  cellGrid:  (7, 7, 7) half cell:  1
```

### PR DESCRIPTION
**Pull request by junghans**
Related #433 

Fix #426

### CODE REVIEW

**Comment by junghans** [comment]
@XzzX 3 builds are failing:

1. Release build on Fedora 36 with gcc 12.1.1: `error: argument 1 null where non-null expected`
2. Documentation build: `LaTeX Error: File `tgtermes.sty' not found.`
3. coverage build: `h5md_parallel ........................***Failed    2.20 sec`, `reference.h5` and `dump.h5` differ

Notes:

- For 1.) the release build fails, but the Debug build is fine????
- For 2.) I will fix that shortly
- For 3.) only the `coverage(push)` builds, but `coverage(pull_request)` is fine, so we should maybe not compare file in binary. Maybe we can use `h5diff` or `h5py` (https://stackoverflow.com/questions/66505315/how-to-compare-multiple-hdf5-files) instead

**Comment by junghans** [comment]
Documentation builds fixed!

**Comment by junghans** [comment]
For 3 this might be a parallel raise condition as we are writing `dump.h5` twice.

**Comment by junghans** [comment]
Ok that was that....

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,9 @@ if(CMAKE_CXX_COMPILER_ID MATCHES GNU)
     if(ESPP_WALL)
         target_compile_options(_espressopp PRIVATE -Wall)
     endif()
+    if(ESPP_WERROR AND CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL 12.1.1 AND CMAKE_BUILD_TYPE STREQUAL "Release")
+        target_compile_options(_espressopp PRIVATE "-Wno-nonnull")
```
Comment (XzzX) [inline src/CMakeLists.txt:31]: I am not sure if I like this. Perhaps it is really an error.

Comment (junghans) [reply src/CMakeLists.txt:31]: I don't like it either but with it being a GCC bug we might need to use this for now.

**Comment by junghans** [comment]
Testing `weekly` CI here: https://github.com/espressopp/espressopp/actions/runs/2636859226

**Comment by junghans** [comment]
Ok, openSUSE has the same issue on gcc-12.1.0

**Comment by junghans** [comment]
Next try for weekly:
https://github.com/espressopp/espressopp/actions/runs/2637503160

**Comment by junghans** [comment]
Ok failure on `WEEKLY`:

1. `ubuntu:devel` with `clang`: `/usr/bin/ld: cannot find -lstdc++: No such file or directory`, this is due to a gcc update 11->12 in progress, so nothing to do here, I saw that in votca as well.
2. `ubuntu:latest` with `mpich`: `h5md_parallel ........................***Exception: SegFault  1.20 sec`, this is https://github.com/espressopp/espressopp/issues/426#issuecomment-1117091339 again

So I guess wait and drop `ubuntu_mpich`.

**Comment by junghans** [comment]
@XzzX please let me know, which part you want to drop and then I can merge this.

**Comment by XzzX** [comment]
Let's drop continue-on-error. The rest is fine.

**Comment by junghans** [comment]
> Let's drop continue-on-error. The rest is fine.

Done, please review.

**Comment by junghans** [comment]
Weekly run here: https://github.com/espressopp/espressopp/actions/runs/2650489225

**Comment by XzzX** [comment]
The h5md test seems still flaky. Perhaps the binary output is non deterministic. I will have a look at it tomorrow an probably change to using h5py, like you suggested.

**Review by XzzX — APPROVED** [review]

**Comment by junghans** [comment]
Let's wait for https://github.com/espressopp/espressopp/actions/runs/2650835748 and then I will merge and have a look at #420.

**Pull request merged**